packages/agent/src/did-api.ts

```diff
@@ -0,0 +1,321 @@
+  TKeyManager = AgentKeyManager,
+  TMethod extends keyof DidMethodCreateOptions<TKeyManager> = keyof DidMethodCreateOptions<TKeyManager>
+> {
+  method: TMethod;
+  options?: DidMethodCreateOptions<TKeyManager>[TMethod];
+  store?: boolean;
+  tenant?: string;
+}
+
+export interface DidMethodCreateOptions<TKeyManager> {
+  dht: DidDhtCreateOptions<TKeyManager>;
+  jwk: DidJwkCreateOptions<TKeyManager>;
+}
+
+export interface DidApiParams {
+  didMethods: DidMethodApi[];
+
+  agent?: Web5PlatformAgent;
+
+  /**
+   * An optional `DidResolverCache` instance used for caching resolved DID documents.
+   *
+   * Providing a cache implementation can significantly enhance resolution performance by avoiding
+   * redundant resolutions for previously resolved DIDs. If omitted, a no-operation cache is used,
+   * which effectively disables caching.
+   */
+  resolverCache?: DidResolverCache;
+
+  store?: AgentDataStore<PortableDid>;
+}
+
+export function isDidRequest<T extends DidInterface>(
+  didRequest: DidRequest<DidInterface>, messageType: T
+): didRequest is DidRequest<T> {
+  return didRequest.messageType === messageType;
+}
+
+export class AgentDidApi<TKeyManager extends AgentKeyManager = AgentKeyManager> extends UniversalResolver {
+  /**
+   * Holds the instance of a `Web5PlatformAgent` that represents the current execution context for
+   * the `AgentDidApi`. This agent is used to interact with other Web5 agent components. It's vital
+   * to ensure this instance is set to correctly contextualize operations within the broader Web5
+   * Agent framework.
+   */
+  private _agent?: Web5PlatformAgent;
+
+  private _didMethods: Map<string, DidMethodApi> = new Map();
+
+  private _store: AgentDataStore<PortableDid>;
+
+  constructor({ agent, didMethods, resolverCache, store }: DidApiParams) {
+    if (!didMethods) {
+      throw new TypeError(`AgentDidApi: Required parameter missing: 'didMethods'`);
+    }
+
+    // Initialize the DID resolver with the given DID methods and resolver cache, or use a default
+    // in-memory cache if none is provided.
+    super({
+      didResolvers : didMethods,
+      cache        : resolverCache ?? new DidResolverCacheMemory()
+    });
+
+    this._agent = agent;
+
+    // If `store` is not given, use an in-memory store by default.
+    this._store = store ?? new InMemoryDidStore();
+
+    for (const didMethod of didMethods) {
+      this._didMethods.set(didMethod.methodName, didMethod);
+    }
+  }
+
+  /**
+   * Retrieves the `Web5PlatformAgent` execution context.
+   *
+   * @returns The `Web5PlatformAgent` instance that represents the current execution context.
+   * @throws Will throw an error if the `agent` instance property is undefined.
+   */
+  get agent(): Web5PlatformAgent {
+    if (this._agent === undefined) {
+      throw new Error('AgentDidApi: Unable to determine agent execution context.');
+    }
+
+    return this._agent;
+  }
+
+  set agent(agent: Web5PlatformAgent) {
```

Comment:
Having both the constructor and the setter feels a little misleading. It's also difficult to understand optionality because up above you see:

```ts
   if (this._agent === undefined) {
      throw new Error('AgentDidApi: Unable to determine agent execution context.');
    }
 ```
 
 Despite the agent property being optional in the class. Right now it's unclear to me whether the agent needs to be provided via constructor, set via setter, or if it's actually unnecessary and will propagate elsewhere